Dockerfile

```diff
@@ -8,6 +8,4 @@ RUN pip install --no-cache-dir --upgrade -r /usr/src/app/requirements.txt
 
 COPY ./app /usr/src/app
 
-ENTRYPOINT ["uvicorn", "app.main:app", "--proxy-headers", "--host", "0.0.0.0"]
-
-CMD ["--port", "8000"]
\ No newline at end of file
+ENTRYPOINT uvicorn app.main:app --proxy-headers --host 0.0.0.0 --port ${NB_API_PORT:-8000}
```

Comment:
1. I feel like the removal of `CMD` keeps showing up in some PRs but I don't think we want that, no? [`CMD`](https://docs.docker.com/engine/reference/builder/#cmd) gives me a default I can easily change. [`ENTRYPOINT`](https://docs.docker.com/engine/reference/builder/#entrypoint) gives me a default I shouldn't have to change. Port is probably something we want to have as a CMD default then
2. `${NB_API_PORT:-8000}` this looks cool - but how does it get in here? I would have expected some `ARG` or `ENV` command to set it first? More importantly: what is it supposed to do? Let you change the default port at build time?
3. I'm assuming there is a reason you changed this line from the `exec` form to the `shell` form - is this linked to the appearance of the shell variable and the disappearance of the CMD?